services/management/backup/backups_service.go

```diff
@@ -0,0 +1,153 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package backup
+
+import (
+	"context"
+
+	backupv1beta1 "github.com/percona/pmm/api/managementpb/backup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// BackupsService represents backups API.
+type BackupsService struct {
+	db          *reform.DB
+	jobsService jobsService
+}
+
+// NewBackupsService creates new backups API service.
+func NewBackupsService(db *reform.DB, jobsService jobsService) *BackupsService {
+	return &BackupsService{
+		db:          db,
+		jobsService: jobsService,
+	}
+}
+
+// StartBackup starts on-demand backup.
+func (s *BackupsService) StartBackup(ctx context.Context, req *backupv1beta1.StartBackupRequest) (*backupv1beta1.StartBackupResponse, error) {
+	var err error
+	var artifact *models.Artifact
+	var location *models.BackupLocation
+	var svc *models.Service
+	var job *models.JobResult
+	var dsn string
+
+	errTX := s.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		svc, err = models.FindServiceByID(tx.Querier, req.ServiceId)
+		if err != nil {
+			return err
+		}
+
+		location, err = models.FindBackupLocationByID(tx.Querier, req.LocationId)
+		if err != nil {
+			return err
+		}
+
+		artifact, err = models.CreateArtifact(tx.Querier, models.CreateArtifactParams{
```

Comment:
I think artifact should be created as result of backup job. at this moment we should create `backup`

Reply:
Do we really need that? If yes, it should be covered here https://confluence.percona.com/pages/viewpage.action?spaceKey=PMM&title=Backup+Management
@atymchuk

Reply:
@BupycHuk that conflicts with architecture. By design, we show artifact right after backup started, so for the user there is no such thing as `backup job`, he sees artifact with some status ("in progress/error/done")

Reply:
ok, let's keep as is, but we should discuss it.

Reply:
Yes, for sure. Let's plan call for that.

Reply:
Discussed